Modules/ST-SEGMENT/st-segment.cpp

```diff
@@ -0,0 +1,295 @@
+#include <memory>
+#include "st-segment.h"
+#include <iostream>
+#include <string>
+
+
+#define FREQUENCY_360 360
+using namespace std;
+
+
+STsegment::STsegment(){}
+
+
+
+vector<int> STsegment::J20points(vector<int> QRSend) {
+// ta funkcja oblicza punkty J20, czyli punkt przesunięty o 20 względem końca załamka QRS
+    vector<int> J20points;
+    for(int i=0; i<QRSend.size(); i++) {
+         J20points.push_back(QRSend[i] + 20);
+
+    }
+    return J20points;
+}
+
+vector<int> STsegment::calculateJpoints(vector<int> J20points , vector<int> Rpeaks){
+// ta funkcja oblicza punkty JX, czyli punkty przesuniete o X wzgledem punktu J20
+    vector<int> JXpoints;
+    int heart_rate,point;
+    float X80 = FREQUENCY_360 * 0.08;
+    float X72 = FREQUENCY_360 * 0.072;
+    float X64 = FREQUENCY_360 * 0.064;
+    float X60 = FREQUENCY_360 * 0.06;
+
+
+    for(int licznik = 0; licznik < J20points.size(); licznik++){
+        heart_rate =  60 / (Rpeaks[licznik] != 0 ? (FREQUENCY_360 * Rpeaks[licznik]) : 1);
+        point = J20points[licznik];
+        if(heart_rate <= 100){
+            JXpoints.push_back(point + X80);
+        }
+        else if (heart_rate> 100 && heart_rate <= 110){
+            JXpoints.push_back(point + X72);
+        }
+        else if (heart_rate > 110 && heart_rate <= 120){
+            JXpoints.push_back(point + X64);
+        }
+        else if (heart_rate > 120){
+            JXpoints.push_back(point + X60);
+        }}
+    return JXpoints;
+}
+
+vector <string> STsegment::calculateOffsets(vector<float> filteredSignal, vector<int> QRSonset, vector<int> JXpoints){
+    float offset;
+    vector <string> Offsets ;
+    for(int i=0;i<JXpoints.size();i++){
+        offset = filteredSignal[JXpoints[i]] - filteredSignal[QRSonset[i]];
+        cout << offset;
```

Comment:
To jest nam nie potrzebne w ostatecznej wersji kodu który będzie mergowany do maina